deepinv/loss/measplit.py

```diff
@@ -21,28 +21,51 @@ class SplittingLoss(Loss):
     By default, the error is computed using the MSE metric, however any other metric (e.g., :math:`\ell_1`)
     can be used as well.
 
+    .. warning::
+
+        The model should be adapted before training using the method :meth:`adapt_model` to include the splitting mechanism at the input.
+
     .. note::
 
         To obtain the best test performance, the trained model should be averaged at test time
         over multiple realizations of the splitting, i.e.
-        :math:`\hat{x} = \frac{1}{N}\sum_{i=1}^N \inversef{y_1^{(i)}}{A_1^{(i)}}`. This can be achieved using
-        :meth:`deepinv.loss.splitting_eval`.
+        :math:`\hat{x} = \frac{1}{N}\sum_{i=1}^N \inversef{y_1^{(i)}}{A_1^{(i)}}`.
 
     :param torch.nn.Module metric: metric used for computing data consistency,
         which is set as the mean squared error by default.
     :param float split_ratio: splitting ratio, should be between 0 and 1. The size of :math:`y_1` increases
         with the splitting ratio.
-    :param bool regular_mask: If ``True``, it will use a regular mask, otherwise it uses a random mask.
+    :param deepinv.physics.generator.PhysicsGenerator, None mask_generator: function to generate the mask. If

```

Comment:
Can we keep the `regular_mask` parameter and deprecate it for backwards compatibility? And can we have a generator for regular masks?

Reply:
I am happy to keep regular_mask for backwards compatibility. However, I think if a regular mask is needed (for denoising) then it is much better to use Neighbor2Neighbor

Reply:
I would actually prefer to get rid of it if that's okay

Reply:
Ok no problem, I'll let our student know